metricbeat/helper/prometheus/prometheus.go

```diff
@@ -213,7 +216,10 @@ func (p *prometheus) ReportProcessedMetrics(mapping *MetricsMapping, r mb.Report
 		return
 	}
 	for _, event := range events {
-		r.Event(mb.Event{MetricSetFields: event})
+		r.Event(mb.Event{
+			MetricSetFields: event,
+			Namespace:       mapping.Namespace,
```

Comment:
Should we set it only in here if it's not empty? Not sure if otherwise it could have some side effects?

Reply:
That's a string, so empty val is "" --> there should be no issue.

Reply:
Just checked the code where the event is generated and indeed we check for `""`